# **📜 Introduction**

**Context**
You have been hired for a new job as a Data Analyst.

The company is called **"TechStream Solutions"**, and the product is a Software as a Service (SaaS) platform named **"Streamline Pro"**. This platform provides comprehensive project management and collaboration tools for businesses of all sizes.

TechStream Solutions has been operating for several years and has gathered significant data on their costs and revenues. They are now looking to analyze their unit economics to understand the profitability of Streamline Pro on a per-customer basis.

The datasets are in the shared folder on Google Drive:
https://drive.google.com/drive/folders/1qhOW9Y2orRXuzbX-kXEmuJ7TMQiRs2Uv?usp=drive_link

**By performing these calculations, TechStream Solutions aims to:**

- Identify the profitability of acquiring and retaining customers.
- Assess the efficiency of their marketing and sales strategies.
- Make informed decisions on scaling their operations and optimizing their resource allocation.
- This information will guide TechStream Solutions in refining their business strategies, ensuring sustainable growth, and maximizing profitability.

**Your First Task: Calculating Unit Economics for Streamline Pro**
**Background:** Streamline Pro is a comprehensive project management and collaboration tool designed to help businesses manage projects, track progress, and collaborate efficiently. Understanding the unit economics of Streamline Pro is crucial for evaluating its financial health and sustainability. This involves analysing key metrics such as Customer Acquisition Cost (CAC), Average Revenue Per User (ARPU), Cost of Goods Sold (COGS), Gross Margin, Customer Lifetime Value (LTV), and the LTV/CAC ratio.

**Objective:** Your task is to calculate the unit economics for Streamline Pro for the month of March 2023. This will help us assess the profitability and efficiency of our customer acquisition strategies and operational expenses.

# **🚀 The Basic Task**

**Task 1:**

In [1]:
import pandas as pd

In [2]:
def load_google_sheet(sheet_id, sheet_name):
    url = f'https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=xlsx'
    return pd.read_excel(url, sheet_name = sheet_name)

receipts_history: https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/edit?gid=952917703#gid=952917703

Payroll: https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/edit?gid=0#gid=0

Monthly expenses: https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/edit?gid=0#gid=0

daily_marketing_spendings: https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/edit?gid=1513891606#gid=1513891606

customer_lifespan_data: https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/edit?gid=1166661477#gid=1166661477

In [3]:
sheets_id = ['1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE',
             '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4',
             '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw',
             '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c',
             '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI']

In [4]:
receipts_history = load_google_sheet(sheets_id[0], 'Sheet1')
Payroll = load_google_sheet(sheets_id[1], 'Sheet1')
Monthly_expenses = load_google_sheet(sheets_id[2], 'Sheet1')
daily_marketing_spendings = load_google_sheet(sheets_id[3], 'Sheet1')
customer_lifespan_data = load_google_sheet(sheets_id[4], 'Sheet1')

In [5]:
receipts_history.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 929 entries, 0 to 928
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   date            929 non-null    datetime64[ns]
 1   customer_id     929 non-null    int64         
 2   receipt_amount  929 non-null    int64         
 3   new_customer    929 non-null    int64         
dtypes: datetime64[ns](1), int64(3)
memory usage: 29.2 KB


In [6]:
receipts_history.head()

date  customer_id  receipt_amount  new_customer
0 2023-01-01         2653              67             1
1 2023-01-01         2731             271             1
2 2023-01-01         1277             231             0
3 2023-01-01         2094             107             0
4 2023-01-01         1314             416             0

In [7]:
number_of_new_customers_acquired = (receipts_history['new_customer'] == 1).sum()
print(number_of_new_customers_acquired)

209


In [8]:
Payroll.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   month          51 non-null     datetime64[ns]
 1   department     51 non-null     object        
 2   employee_name  51 non-null     object        
 3   position       51 non-null     object        
 4   paid           51 non-null     int64         
dtypes: datetime64[ns](1), int64(1), object(3)
memory usage: 2.1+ KB


In [9]:
Payroll = Payroll.convert_dtypes()

In [10]:
Payroll.head()

month department  employee_name           position  paid
0 2023-01-01      Sales       John Doe      Sales Manager  1500
1 2023-01-01      Sales     Jane Smith    Sales Associate   600
2 2023-01-01      Sales      Jim Brown    Sales Associate   700
3 2023-01-01      Sales   Laura Miller    Sales Associate   800
4 2023-01-01  Marketing  Alice Johnson  Marketing Manager  1650

In [11]:
Payroll['department'].unique()

<StringArray>
['Sales', 'Marketing', 'Engineering', 'HR', 'Support', 'Analytics']
Length: 6, dtype: string

In [12]:
sales_expenses = Payroll[(Payroll['month'].dt.strftime('%Y-%m') == '2023-03') & (Payroll['department'].isin(['Sales','Marketing']))]['paid'].sum()
print(sales_expenses)

5950


In [13]:
Monthly_expenses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   #         27 non-null     int64         
 1   month     27 non-null     datetime64[ns]
 2   category  27 non-null     object        
 3   item      27 non-null     object        
 4   amount    27 non-null     int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 1.2+ KB


In [ ]:
Monthly_expenses = Monthly_expenses.convert_dtypes()

In [14]:
Monthly_expenses.head()

#      month           category                  item  amount
0  1 2023-01-01       Server Costs           AWS Hosting    8000
1  2 2023-01-01       Server Costs  Google Cloud Storage    4000
2  3 2023-01-01  Software Licenses        Atlassian Jira    1200
3  4 2023-01-01  Software Licenses                 Slack     800
4  5 2023-01-01  Software Licenses            Salesforce    1500

In [15]:
Monthly_expenses['category'].unique()

array(['Server Costs', 'Software Licenses', 'Office Rental', 'Other'],
      dtype=object)

In [16]:
crm_cost = Monthly_expenses[Monthly_expenses['month'].dt.strftime('%Y-%m') == '2023-03']['amount'].sum()
print(crm_cost)

31140


In [17]:
daily_marketing_spendings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   date      360 non-null    datetime64[ns]
 1   channel   360 non-null    object        
 2   spending  360 non-null    int64         
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 8.6+ KB


In [18]:
daily_marketing_spendings = daily_marketing_spendings.convert_dtypes()

In [19]:
daily_marketing_spendings.head()

date       channel  spending
0 2023-01-01    Google Ads       784
1 2023-01-01  Facebook Ads       659
2 2023-01-01  LinkedIn Ads       729
3 2023-01-01   Twitter Ads       292
4 2023-01-02    Google Ads       935

In [20]:
daily_marketing_spendings['channel'].unique()

<StringArray>
['Google Ads', 'Facebook Ads', 'LinkedIn Ads', 'Twitter Ads']
Length: 4, dtype: string

In [21]:
marketing_expenses = daily_marketing_spendings[daily_marketing_spendings['date'].dt.strftime('%Y-%m') == '2023-03']['spending'].sum()
print(marketing_expenses)

68830


In [22]:
customer_lifespan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   Unnamed: 0  100 non-null    int64         
 1   start_date  100 non-null    datetime64[ns]
 2   churn_date  100 non-null    datetime64[ns]
dtypes: datetime64[ns](2), int64(1)
memory usage: 2.5 KB


In [23]:
customer_lifespan_data.head()

Unnamed: 0 start_date churn_date
0        1000 2021-11-15 2022-09-14
1        1001 2022-04-15 2023-02-16
2        1002 2022-10-30 2023-02-04
3        1003 2021-08-22 2023-02-07
4        1004 2021-08-23 2022-02-02

In [24]:
CAC = (crm_cost + sales_expenses + marketing_expenses)/number_of_new_customers_acquired
print(CAC)

506.79425837320576


In [25]:
revenue = receipts_history[receipts_history['date'].dt.strftime('%Y-%m') == '2023-03']['receipt_amount'].sum()
print(revenue)

83033


In [26]:
ARPU = receipts_history[receipts_history['date'].dt.strftime('%Y-%m') == '2023-03'].groupby('customer_id')['receipt_amount'].sum().mean()
print(ARPU)

284.3595890410959


In [27]:
COGS = Monthly_expenses.loc[(Monthly_expenses['month'].dt.strftime('%Y-%m') == '2023-03') & (Monthly_expenses['category'].isin(['Server Costs', 'Software Licenses']))]['amount'].sum()
print(COGS)

17340


In [28]:
gross_margin = (revenue - COGS)/revenue
print(gross_margin)

0.7911673671913576


In [29]:
average_customer_lifespan = (customer_lifespan_data['churn_date'] - customer_lifespan_data['start_date']).mean().total_seconds()/(30.44 * 24 * 3600)
print(average_customer_lifespan)

9.699080157687254


In [30]:
LTV = ARPU * average_customer_lifespan * gross_margin
print(LTV)

2182.060523284071


In [31]:
print(LTV/CAC)

4.305614136767097


In [32]:
unit_economics = {'CAC': CAC,
                  'ARPU': ARPU,
                  'COGS': COGS,
                  'Gross Margin(%)': [gross_margin*100],
                  'LTV': LTV,
                  'LTV/CAC': LTV/CAC
                  }
unit_economics = pd.DataFrame(unit_economics)
unit_economics = unit_economics.melt(var_name='Metric', value_name='Value')
print(unit_economics)

            Metric         Value
0              CAC    506.794258
1             ARPU    284.359589
2             COGS  17340.000000
3  Gross Margin(%)     79.116737
4              LTV   2182.060523
5          LTV/CAC      4.305614


**Task 2:**

**✅ Conclusion:**
- CAC (Customer Acquisition Cost) is $506.79 – relatively high, indicating the company is heavily investing in customer acquisition (advertising, sales, onboarding). If not optimized, this may hurt profitability.

- ARPU (Average Revenue Per User) is $284.36, reflecting a solid average income per user. However, given the high CAC, it's crucial to increase the value derived from each customer.

- COGS (Cost of Goods Sold) is $17,340, which typically includes server costs, technical support, maintenance, etc. This appears high compared to ARPU, suggesting a need to optimize operational costs.

- Gross Margin is 79.12%, which is healthy and aligns with SaaS industry standards (above 70%) – indicating a profitable business model.

- LTV (Customer Lifetime Value) is $2,182.06, showing that each customer brings substantial value over their lifecycle.

- LTV/CAC ratio is 4.31, exceeding the benchmark of 3.0 – indicating strong efficiency in monetizing acquired customers and potential for sustainable growth.

**💡 Recommendations:**
- **Maintain LTV/CAC > 3.0:** Your current ratio of 4.31 is excellent. Maintain this by improving customer retention and reducing acquisition costs.

- **Increase ARPU by:**

 - Upselling: Offer premium features or higher-tier plans.

 - Cross-selling: Provide related services (e.g., analytics, security, API integrations).

- **Optimize CAC:**

 - Invest in inbound marketing (content, SEO, referrals),

 - Automate onboarding processes,

 - Shift towards a PLG (Product-Led Growth) strategy.

- **Reduce COGS where possible:**

 - Optimize cloud infrastructure,

 - Automate customer support,

 - Reduce reliance on manual support efforts.

- **Extend Customer Lifetime** through proactive customer success strategies, robust documentation, educational webinars, and fast technical support to minimize churn.

- **Monitor churn rate regularly:** Since churn directly impacts LTV, it should be tracked and improved continuously.